src/ceph-daemon/ceph-daemon

```diff
@@ -256,7 +299,10 @@ def makedirs(dir, uid, gid, mode):
 
 def get_data_dir(fsid, t, n):
     # type: (str, str, Union[int, str]) -> str
-    return os.path.join(args.data_dir, fsid, '%s.%s' % (t, n))
+    if n:
+        return os.path.join(args.data_dir, fsid, '%s.%s' % (t, n))
+    else:
+        return os.path.join(args.data_dir, fsid, '%s' % (t))
```

Comment:
we could name the prometheus instances like ``prometheus.cxwads`` like we do the mgr etc and avoid this weirdness...

Reply:
what about accepting prometheus as the name, but in deploy and remove, if the id is None, set it to 0 - then everything else stays the same?

Reply:
@liewegas updated to use <service>.<hostname> so the dir for the monitoring components will look more like mon/mgr dirs